src/gen/conditions.rs

```diff
@@ -0,0 +1,1643 @@
+        }
+        ASSERT_PUZZLE_ANNOUNCEMENT => {
+            let id = sanitize_hash(
+                a,
+                first(a, c)?,
+                32,
+                ErrorCode::AssertPuzzleAnnouncementFailed,
+            )?;
+            Ok(Condition::AssertPuzzleAnnouncement(id))
+        }
+        ASSERT_MY_COIN_ID => {
+            let id = sanitize_hash(a, first(a, c)?, 32, ErrorCode::AssertMyCoinIdFailed)?;
+            Ok(Condition::AssertMyCoinId(id))
+        }
+        ASSERT_MY_PARENT_ID => {
+            let id = sanitize_hash(a, first(a, c)?, 32, ErrorCode::AssertMyParentIdFailed)?;
+            Ok(Condition::AssertMyParentId(id))
+        }
+        ASSERT_MY_PUZZLEHASH => {
+            let id = sanitize_hash(a, first(a, c)?, 32, ErrorCode::AssertMyPuzzlehashFailed)?;
+            Ok(Condition::AssertMyPuzzlehash(id))
+        }
+        ASSERT_MY_AMOUNT => {
+            let amount = parse_amount(
+                a,
+                first(a, c)?,
+                ErrorCode::AssertMyAmountFailed,
+                range_cache,
+                flags,
+            )?;
+            Ok(Condition::AssertMyAmount(amount))
+        }
+        ASSERT_SECONDS_RELATIVE => Ok(Condition::AssertSecondsRelative(parse_seconds(
+            a,
+            first(a, c)?,
+            ErrorCode::AssertSecondsRelative,
+            range_cache,
+            flags,
+        )?)),
+        ASSERT_SECONDS_ABSOLUTE => Ok(Condition::AssertSecondsAbsolute(parse_seconds(
+            a,
+            first(a, c)?,
+            ErrorCode::AssertSecondsAbsolute,
+            range_cache,
+            flags,
+        )?)),
+        ASSERT_HEIGHT_RELATIVE => Ok(Condition::AssertHeightRelative(parse_height(
+            a,
+            first(a, c)?,
+            ErrorCode::AssertHeightRelative,
+            range_cache,
+            flags,
+        )?)),
+        ASSERT_HEIGHT_ABSOLUTE => Ok(Condition::AssertHeightAbsolute(parse_height(
+            a,
+            first(a, c)?,
+            ErrorCode::AssertHeightAbsolute,
+            range_cache,
+            flags,
+        )?)),
+        _ => Err(ValidationErr(c, ErrorCode::InvalidConditionOpcode)),
+    }
+}
+
+#[derive(Debug)]
+pub struct SpendConditionSummary {
+    pub coin_id: Arc<[u8; 32]>,
+    pub puzzle_hash: NodePtr,
+    // conditions
+    // The sum of all reserve fee conditions
+    pub reserve_fee: u64,
+    // the highest height/time conditions (i.e. most strict)
+    pub height_relative: u32,
+    pub height_absolute: u32,
+    pub seconds_relative: u64,
+    pub seconds_absolute: u64,
+    // all create coins. Duplicates are consensus failures
+    pub create_coin: HashSet<(Vec<u8>, u64)>,
+    // Agg Sig conditions
+    pub agg_sigs: Vec<Condition>,
+}
+
+struct AnnounceState {
+    // hashing of the announcements is deferred until parsing is complete. This
+    // means less work up-front, in case parsing/validation fails
+    coin: HashSet<(Arc<[u8; 32]>, NodePtr)>,
```

Comment:
Maybe `announcement_by_coin_id`? Is that too long? I feel like `coin` doesn't quite capture this. Same with `puzzle`

Reply:
I figured since they are already part of `AnnounceState`, it's implied that they are announcements.